ivre/db/mongo.py

```diff
@@ -1549,6 +1550,24 @@ def migrate_schema_hosts_17_18(doc):
             update["$set"]["ports"] = doc['ports']
         return update
 
+    @staticmethod
+    def migrate_schema_hosts_18_19(doc):
+        """Converts a record from version 18 to version 19. Version 19
+splits smb-os-discovery scripts into two, a smb-ntlm-info one that contains all
+the information the original smb-os-discovery script got from NTLM, and a
+smb-os-discovery script with only the information regarding SMB
+
+        """
+        assert doc["schema_version"] == 18
+        doc["schema_version"] = 19
+        for port in doc.get('ports', []):
+            for i, script in enumerate(port.get('scripts', [])):
+                if script['id'] == 'smb-os-discovery':
+                    smb, ntlm = xmlnmap.split_smb_os_discovery(script)
+                    port.get('scripts', [])[i] = smb
+                    port.get('scripts', []).append(ntlm)
+        return doc
```

Comment:
This will not work, as you need to return an `update` document.
```suggestion
        update = {"$set": {"schema_version": 19}}
        updated = False
        for port in doc.get('ports', []):
            for script in port.get('scripts', []):
                if script['id'] == 'smb-os-discovery':
                    smb, ntlm = xmlnmap.split_smb_os_discovery(script)
                    script.update(smb)
                    port['scripts'].append(ntlm)
                    updated = True
        if updated:
            update["$set"]["ports"] = doc['ports']
        return update
```